### DISCUSSION TITLE
ESP32 connection via RShell/REPL issues

### DISCUSSION THREAD

**Discussion by krishnak** [micropython/micropython#9477] [Hardware & Peripherals]
I have three ESP32 board which are classified as Node MCU 32s. When I flash a ESP-IDF based C code in them and then connect using monitor - they all connect each and every time to the serial port and function without any issues.

When Micropython v1.19 is flashed on the device the following happens.

1. REPL takes ages to get board name or fails to gethostname and throws errors.
2.  The actual issue of 1 is - the firmware has crashed and is rebooting and further crashing in the background.
3. This behaviour is "ELIMINATED" only if you have the following lines in the boot.py

```import network
   network.WLAN(network.AP_IF).active(True)
```
If these lines are absent - it becomes impossible to get in to REPL to transfer files.

4. However if these lines are absent AND if you have a main.py code - it executes without any issues on boot (if you are not trying to connect via Rshell)
So the issue occurs only when serial port is trying to connect - this may be related to the use of UART by REPL - can you please investigate.

**Comment by krishnak** [comment]
I have isolated this problem from the user code. When at REPL prompt, if some of the pins are touched with finger or if they touch on any other object this reset occurs on ESP32, I tracked it down to a bug in ESP-IDF - which appears to have been fixed in 2019, the error on screen is as follows, is there a way to disable interrupts to see whether it is a cause

```
icroPython v1.19-espnow-krishnak on 2022-10-01; ESP32 module with ESP32
Type "help()" for more information.
>>> ets Jun  8 2016 00:22:57

rst:0x8 (TG1WDT_SYS_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4396
ho 0 tail 12 room 4
load:0x40078000,len:13884
load:0x40080400,len:3340
entry 0x40080618
W (57) boot.esp32: PRO CPU has been reset by WDT.
W (58) boot.esp32: WDT reset info: PRO CPU PC=0x400816a1
W (59) boot.esp32: WDT reset info: APP CPU PC=0x400816a1
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:4396
ho 0 tail 12 room 4
load:0x40078000,len:13884
load:0x40080400,len:3340
entry 0x40080618
MicroPython v1.19-espnow-krishnak on 2022-10-01; ESP32 module with ESP32
Type "help()" for more information.

```

**Comment by krishnak** [comment]
Again this doesn't occur when a standard C program using ESP_IDF is run - i.e not sensitive to touching the board pins or anyother issues

**Comment by robert-hh** [comment]
Does that happen with one of the standard firmware versions from the download site as well?

**Reply by krishnak** [reply]
yes

**Reply by krishnak** [reply]
It is a combination of UART being connected and few other things that causes it. If you don't connect to the serial port and if you are I use a main.py - then the code works without issues.

In all the boards, if I remove the code that I have added to bring up the wifi interface up in the boot.py (see first message) - then serial port connection will crash immediately after entering micropython prompt with this error.

With the boot.py it boots up and everything works alright, unless some pins are touched. 
In one board it is very sensitive, touching any pin causing the crash is reproduceable everytime- others too have this but more spordaic.
